### PR TITLE
Microsoft graph mail- add page size limit

### PR DESCRIPTION
**Pull request by tallieber**
## Status
- [x] Ready

## Related Issues
reference: https://github.com/demisto/etc/issues/33738

## Description
Add page size limit to the request to avoid timeout.

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.yml [hunk 1]
```diff
@@ -92,6 +92,18 @@ script:
...
+      isArray: false
+      name: timeout
+      required: false
+      secret: false
+    - default: false
+      description: 'Limit emails to fetch in one request'
+      isArray: false
+      name: page_size
+      required: false
+      secret: false
```
Comment (bakatzir) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.yml:106]: I suggest adding a default value of 20

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py [hunk 2]
```diff
@@ -109,13 +109,22 @@ def list_mails(self, user_id: str, folder_id: str = '', search: str = None, odat
         no_folder = f'/users/{user_id}/messages'
         with_folder = f'/users/{user_id}/{build_folders_path(folder_id)}/messages'
         pages_to_pull = demisto.args().get('pages_to_pull', 1)
+        user_timeout = float(demisto.args().get('timeout', 300))
```
Comment (bakatzir) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py:112]: after adding the paging, I do not think the timeout arg is needed

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py [hunk 3]
```diff
@@ -1112,14 +1121,26 @@ def list_mails_command(client: MsGraphClient, args):
     odata = args.get('odata')
 
     raw_response = client.list_mails(user_id, folder_id=folder_id, search=search, odata=odata)
+    last_page_response = raw_response[len(raw_response)-1]
+    metadata = ''
+    next_page = last_page_response.get('@odata.nextLink')
+    if next_page:
+        metadata = f'\nPay attention there are more results than shown. For more data please ' \
+                               f'increase "pages_to_pull" argument'
```
Comment (bakatzir) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py:1129]: by how much? do you get the total amounts of mails?
we can discuss it, but in general, we prefer that each time the cmd run x(20) mails will be pulled.
so e.g:
`!ms-graph-mail-list-emails pages_to_pull=1 page_size =2` - will return the first 20.
`!ms-graph-mail-list-emails pages_to_pull=2 page_size =2` - will return the 20 to 40 emails.

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.yml [hunk 4]
```diff
@@ -92,6 +92,18 @@ script:
       name: search
       required: false
       secret: false
+    - default: false
+      description: 'Request timeout in seconds'
+      isArray: false
+      name: timeout
+      required: false
+      secret: false
```
Comment (bakatzir) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.yml:100]: again, let's re think if this is needed

**Review by bakatzir — COMMENTED** [review]
Please add tasks to the TPB with different pages numbers and validate the amount of results.

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py (hunk 3 above)
Comment (tallieber) [reply Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py:1129]: I dont get the total amount of results, and your example is exactly the logic, pages_to_pull argument implemented before my change and this is exactly what he do- it loop over the number of pages and concatenate

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.yml (hunk 4 above)
Comment (tallieber) [reply Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.yml:100]: agree, is not needed

**Review by bakatzir — APPROVED** [review]

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.yml [hunk 5]
```diff
@@ -101,7 +107,8 @@ script:
       required: false
       secret: false
     deprecated: false
-    description: Gets the properties of returned emails.
+    description: Gets the properties of returned emails. Usually shows partialy results,
+      use the "page_size" and "pages_to_pull" arguments to get all results.
```
Comment (amshamah419) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.yml:111]:
```suggestion
    description: Gets the properties of returned emails. Typically shows partial results,
      use the "page_size" and "pages_to_pull" arguments to get all results.
```

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/README.md [hunk 6]
```diff
@@ -78,10 +79,10 @@ Gets the properties of returned emails.
 | MSGraphMail.SendTime | Date | The time the email was sent. | 
 | MSGraphMail.Categories | String | Categories of the email. | 
 | MSGraphMail.HasAttachments | Boolean | Whether the email has attachments. | 
-| MSGraphMail.Subject | String | The subject of the email. | 
+| MSGraphMail.Subject | String | The subject of email. | 
 | MSGraphMail.IsDraft | Boolean | Whether the email is a draft. | 
-| MSGraphMail.Body | String | The content (body) of the email. | 
-| MSGraphMail.Sender.Name | String | The name of the sender. | 
+| MSGraphMail.Body | String | The content \(body\) of the email. | 
+| MSGraphMail.Sender.Name | String | The name of sender. |
```
Comment (amshamah419) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/README.md:85]:
```suggestion
| MSGraphMail.Sender.Name | String | The name of the sender. | 
```

File: Packs/MicrosoftGraphMail/ReleaseNotes/1_0_18.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Microsoft Graph Mail
+- Add page size argument to **msgraph-mail-list-emails** to limit the number of emails in each request.
```
Comment (amshamah419) [inline Packs/MicrosoftGraphMail/ReleaseNotes/1_0_18.md:4]:
```suggestion
- Added the *page_size* argument to the  ***msgraph-mail-list-emails*** in order to limit the number of emails in each request.
```

File: Packs/MicrosoftGraphMail/ReleaseNotes/1_0_18.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+- Add NextPage attribute to **msgraph-mail-list-emails** outputs to retrieve additional emails if needed.
```
Comment (amshamah419) [inline Packs/MicrosoftGraphMail/ReleaseNotes/1_0_18.md:5]:
```suggestion
- Added the *NextPage* attribute to the  ***msgraph-mail-list-emails*** command outputs in order to retrieve additional emails if needed.
```

**Pull request merged**